src/main/java/hudson/remoting/jnlp/Main.java

```diff
@@ -264,6 +268,38 @@ public static void _main(String[] args) throws IOException, InterruptedException
         if(m.urls.isEmpty() && m.directConnection == null) {
             throw new CmdLineException(p, "At least one -url option is required.", null);
         }
+        if (m.webSocket) {
+            if (m.urls.isEmpty()) {
+                throw new CmdLineException(p, "-url is required in -webSocket mode", null);
+            }
+            if (m.urls.size() > 1) {
+                throw new CmdLineException(p, "multiple -url is not currently supported in -webSocket mode", null);
+            }
+            if (m.directConnection != null) {
+                throw new CmdLineException(p, "-webSocket and -direct are mutually exclusive", null);
+            }
+            if (m.tunnel != null) {
+                throw new CmdLineException(p, "-tunnel is not currently supported in -webSocket mode", null);
```

Comment:
I suggest dropping the word "currently" from this and the following messages. It's not clear why we would ever add support for some of these. If it comes up then we can change things anyway.

You could use the `forbids` attribute from args4j for each of these conflicting options, but this way also works.

Reply:
> It's not clear why we would ever add support for some of these.

I used _currently_ when there was some plausible reason why we might want such a combination. In the case of `-tunnel`, I had thought this would apply even to the HTTP metadata connection, but a closer reading of the TCP-mode code suggests that this is not so, so I will fix that.

> You could use the `forbids` attribute from args4j

Neat, I did not know about that.

Reply:
The tunnel one is particularly one we don't want to propagate. It's problematic as it is.